sklearn/model_selection/_split.py

```diff
@@ -745,7 +745,15 @@ def __init__(self, n_splits=5, *, shuffle=False, random_state=None):
 
     def _make_test_folds(self, X, y=None):
         rng = check_random_state(self.random_state)
-        y = np.asarray(y)
+        # XXX: as of now, cross-validation splitters only operate in NumPy-land
+        # without attempting to leverage array API namespace features. However
+        # they might be fed by array API inputs, e.g. in CV-enabled estimators so
+        # we need the following explicit conversion:
+        xp, is_array_api = get_namespace(y)
+        if is_array_api:
+            y = _convert_to_numpy(y, xp)
```

Comment:
Would it be possible to add a test to cover this?

Reply:
I think we had coverage for it via the test I removed in the last commit. But I thought that we tested the same thing via a common test. So I am a bit puzzled why that doesn't happen :-/

Reply:
I understand what's going on:

- the common tests run the search cv meta estimator using `LogisticRegression` and `Ridge` as base-estimator.
- the `LogisticRegression` test is skipped because `LogisticRegression` does not have the array API estimator tag hence the wrapping search cv estimators ain't either;
- the stratified k-fold CV splitter is only used for classification problem, hence the `Ridge`-based common test does not cover it;
- the previous hard-coded test of this PR used `LinearDiscriminantAnalysis` which is a classifier and supports array API, hence could cover this line.

Since adding array API support to `LogisticRegression` might take a bit of time, I would be in favor of re-adding the previous non-common test that was removed from this PR.

Reply:
Yeah that seems right. Thanks for the explanation! I think the test seems valid for now and not redundant.

Reply:
@ogrisel I added the test back. Can you have a look and then we can probably merge.